### PR TITLE
Update SPA JS, TS and add orgs example

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### Description

- Update SPA JS
- Make sure SPA JS organization docs appear in auth0-react docs (update TS and tsdoc)
- Add an organizations example

### References

image

### CODE REVIEW

File: src/auth0-context.tsx [hunk 1]
```diff
@@ -13,24 +12,15 @@ import {
 import { createContext } from 'react';
 import { AuthState, initialAuthState } from './auth-state';
 
-export interface RedirectLoginOptions extends BaseLoginOptions {
+export interface RedirectLoginOptions
```
Comment (adamjmcgrath) [inline src/auth0-context.tsx:15]: The new version of tsdoc is better at surfacing inherited types

File: src/with-auth0.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ const withAuth0 = <P extends WithAuth0Props>(
 ): ComponentType<Omit<P, keyof WithAuth0Props>> => (props): JSX.Element => (
   <Auth0Context.Consumer>
     {(auth: Auth0ContextInterface): JSX.Element => (
-      <Component auth0={auth} {...(props as P)} />
+      <Component {...(props as P)} auth0={auth} />
```
Comment (adamjmcgrath) [inline src/with-auth0.tsx:31]: The new version of TS was grumbling at the order of these props

File: EXAMPLES.md [hunk 3]
```diff
@@ -260,3 +261,41 @@ export const Profile = () => {
   );
 };
 ```
+
+## 5. Accept a user invite and login to an organization
```
Comment (frederikprijck) [inline EXAMPLES.md:265]: Should we have guidance around logging in to an org without accepting a user invitation? I think both SPAJS and Angular SDK treat these separately from one another.

File: src/auth0-provider.tsx [hunk 4]
```diff
@@ -29,7 +28,8 @@ export type AppState = {
 /**
  * The main configuration to instantiate the `Auth0Provider`.
  */
-export interface Auth0ProviderOptions {
+export interface Auth0ProviderOptions
+  extends Exclude<Auth0ClientOptions, 'client_id' | 'redirect_uri'> {
```
Comment (frederikprijck) [inline src/auth0-provider.tsx:32]: TIL about Exclude<...> 😱

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Comment by adamjmcgrath** [comment]
Just gonna close while I investigate these test failures

**Comment by adamjmcgrath** [comment]
Reverted the TS updates (need to wait for TS 3.9)

File: EXAMPLES.md (hunk 3 above)
Comment (adamjmcgrath) [reply EXAMPLES.md:265]: GH wouldn't let me ropen this one - have raised a new one at #221 

> Should we have guidance around logging in to an org without accepting a user invitation?

Have added some more comments around the `organization` init argument - I'd like to keep all the guidance in one example

**Pull request closed without merging**